subsys/usb/class/cdc_acm.c

```diff
@@ -236,7 +236,7 @@ int cdc_acm_class_handle_req(struct usb_setup_packet *pSetup,
 		break;
 
 	case SET_CONTROL_LINE_STATE:
-		dev_data->line_state = (uint8_t)sys_le16_to_cpu(pSetup->wValue);
+		dev_data->line_state = (uint8_t)(pSetup->wValue & 0xFF);
```

Comment:
Neither the cast to `(uint8_t)` nor the `& 0xFF` are necessary. The compiler will do the right thing. I am ok if you want to leave those, but just as an FYI.

Reply:
Yes I know but I wanted to underline that. Same for all other around the place.

Reply:
If there would be more votes to remove that - no problem.

Reply:
agree with @carlescufi, IMHO it is hard to read

Reply:
+1 to it being more readable as an implicit cast.

Reply:
for sure not, how is `(uint8_t)(pSetup->wValue & 0xFF);` more readable? specially if it is not necessary?

Reply:
Sorry @jfischer-phytec-iot by implicit cast I mean:

dev_data->line_state = pSetup->wValue

There's no explicit cast as the compiler does it for us.

I'm *not* suggesting using the (uint8_t)(foo & mask) version.

Reply:
@carlescufi I agree that `& 0xFF` is redundant in this case. But casting to `uint8_t` is not. Of course, the compiler will manage to apply the proper conversion by itself, but it can also issue a warning for such assignment (see `-Wconversion` in gcc), what I think is undesirable.

Reply:
@anangl thanks for the info, I didn't know the compiler could actually issue a warning for this. In this case I think it makes sense that we just remove the `& 0xFF`.